classes/local/lib.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_cms\local;
+
+use core_course\local\entity\content_item;
+use core_course\local\entity\string_title;
+use mod_cms\local\model\cms;
+use moodle_url;
+use stdClass;
+
+/**
+ * Generic library functions for mod_cms.
+ *
+ * @package   mod_cms
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2023, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class lib {
+    /** @var string The URL for editing a mod instance. */
+    protected const MODEDIT_URL = '/course/modedit.php';
+
+    /**
+     * Obtains a list of defined content types to be included in the activity chooser panel.
+     *
+     * @param content_item $defaultmodulecontentitem
+     * @param \stdClass $user Not used.
+     * @param \stdClass $course Not used.
+     * @return array
+     */
+    public static function get_course_content_items(content_item $defaultmodulecontentitem, stdClass $user,
+        stdClass $course) : array {
+        global $COURSE;
+
+        $items = [];
+
+        $baseurl = $defaultmodulecontentitem->get_link();
+        $linkurl = new moodle_url(
+            self::MODEDIT_URL,
+            ['id' => $baseurl->param('id'), 'course' => $COURSE->id, 'add' => 'cms']
+        );
+
+        $types = model\cms_types::get_records();
```

Comment:
For consistency can we define a `use mod_cms\local\model\cms_types;` and then
```suggestion
        $types = cms_types::get_records();
```